### ISSUE TITLE
CSRs in VexRiscv are violating the priv spec

### ISSUE THREAD

**Issue by pgielda** [enjoy-digital/litex#87]
VexRiscv defines two registers that were added as a plugin specially for LiteX:
```
csr.rw(0x330, mask)
csr.r(0x360, pendings)
```
In my opinion both violate the priv spec (both 1.9 and 1.10).
0x330 is the most problematic as it violates this definition:
```
#define CSR_MHPMEVENT16 0x330
```
It is easy to spot by issuing this:
```
riscv64-unknown-elf-objdump -S bios.elf | grep mhpmevent16
```
While I understand that you could always say that a cpu does not follow priv spec at all, I would still suggest to follow it at least to a point of using ranges that are meant for custom CSRs (see *Table 2.1: Allocation of RISC-V CSR address ranges.* on page 8 in the spec: https://people.eecs.berkeley.edu/~krste/papers/riscv-privileged-v1.9.1.pdf).

There are two benefits of that:
- disassembly using a standard riscv gcc will make more sense
- it is easier to write simulators for a core like that (<-- main reason why I've discovered this at all).

In other words I would move those two CSRs to *Non-standard read/write* and *Non-standard read only* areas respectively.

I already wrote to Charles Papon (author of VexRiscv) about this but I think it makes sense to point it out here as well as this will have to be updated both in VexRiscv plugin and in the LiteX code.

**Comment by enjoy-digital** [comment]
Thanks. The best is probably indeed to update VexRiscv on this point and then update LiteX code.
Do you know if Charles is going to do the changes? Can you update this issue when changes have landed in Vexriscv?

**Comment by pgielda** [comment]
I will discuss that with Charles over email and see if he is going to do it.

**Comment by enjoy-digital** [comment]
Thanks.

**Comment by pgielda** [comment]
Charles wrote he should do it soon.

**Comment by pgielda** [comment]
https://github.com/Dolu1990/misoc/commit/2441dc0f433e2c671be1be00ad25e649347fad7d
https://github.com/m-labs/VexRiscv-verilog/commit/2b498dcf67528d4c634d7b2247d7400837b95d30#diff-2b4b27e9af32a902c3cd4c8ff9acc7c4R111

**Comment by enjoy-digital** [comment]
Thanks, it's updated.
